azure/scope/managedcontrolplane.go

```diff
@@ -500,6 +500,7 @@ func (s *ManagedControlPlaneScope) GetAgentPoolSpecs(ctx context.Context) ([]azu
 			Replicas:          1,
 			OSDiskSizeGB:      0,
 			Mode:              pool.Spec.Mode,
+			MaxPods:           pool.Spec.MaxPods,
```

Comment:
this is honestly easy to miss and it's not very surprising that a bug like this was introduced. I can easily see more similar bugs happening in the future unless reviewers are extra careful. Doesn't have to be in this PR, but it would be great if we can find a way to make this code a bit less error-prone, maybe by extracting the common logic from these two functions, thoughts?

Reply:
yes, I can take a stab on separate PR as we are getting different specs of agentpool from two different methods. Ideally, we should get only one same spec. 

As there are other fields that needs to be tested carefully like `VnetSubnetID`, `ResourceGroup` or clustername etc